### ISSUE TITLE
Add text-to-speech feature to webbook

### ISSUE THREAD

**Issue by josieg** [pressbooks/ideas#224]
### Feature Description
Provide people reading a book via the Pressbooks webbook the option to have the content on a page/chapter read aloud to them without having to install software or a plugin. 

### Feature Use Case
This would would help increase the accessibility of resources published in Pressbooks. It would benefit all students, but it would specifically benefit students who have various vision impairments, students with learning and attention disabilities, and students who prefer listening to content for any reason.

This idea is related to #57 (which suggests being able to export audio files).

### Other Notes

I found this WordPress plugin that might be a possibility??: [ResponsiveVoice Text to Speech](https://wordpress.org/plugins/responsivevoice-text-to-speech/)

**Comment by SteelWagstaff** [comment]
@josieg suggested via Twitter that we may also wish to consider https://trinityaudio.ai/the-trinity-audio-wordpress-plugin-implementation-guide/. See implementation by Clint Lalonde for WP blog at https://edtechfactotum.com/could-a-canadian-mooc-provider-have-helped-higher-ed-this-fall/

@ricardopressbooks Has suggested we could perhaps try to integrate the [AWS Polly service](https://aws.amazon.com/polly/pricing/?nc1=h_ls) in order to allow users to listen to audio from books (chapters or any text, to define) through some custom plugin or use [the official AWS Polly WP plugin](https://wordpress.org/plugins/amazon-polly/). Tutorial: https://www.wpbeginner.com/plugins/how-to-add-text-to-speech-in-wordpress/

There is also interest from @beckej13820 in text to speech as an export format.

**Comment by josieg** [comment]
Would you believe I forgot I submitted this? Thanks Steel!

**Comment by SteelWagstaff** [comment]
@josieg it was a good idea then (and now). Thanks for keeping up your vigilance for things that would make Pressbooks more accessible to more users!

**Comment by pb-amy** [comment]
University of Houston sent forth the same suggestion -- https://pressbooks.zendesk.com/agent/tickets/7075. Would be a great feature to explore!

**Comment by tw77** [comment]
University of Minnesota also requests this: https://pressbooks.zendesk.com/agent/tickets/9648

**Comment by SteelWagstaff** [comment]
Still relevant.

**Comment by arzola** [comment]
This issue has been automatically closed as part of an organization-wide backlog review.

**Reason:** This issue was created 6y 4m ago with no updates in the last 8m.

If this is still relevant, please feel free to reopen it with updated context. We appreciate your contributions and want to ensure our issue tracker stays actionable.

**Learn more:** https://pressbooks.community/t/announcement-automatic-closure-of-old-github-tickets/4542

**Comment by SteelWagstaff** [comment]
Joel Gladd has written about how he's been doing this for his books in Pressbooks using a Claude Code skill: https://joelgladd.substack.com/p/how-im-turning-my-textbook-into-audio + https://github.com/Brehove/tts-toolkit

**Comment by MichSpieler** [comment]
Similar request: https://github.com/pressbooks/ideas/issues/530
Users would like to be able to add SSML markup to give TTS tools indications on pronunciation.

**Comment by greatislander** [comment]
Worth noting that many user agents on macOS and Windows now provide this functionality out of the box.

E.g. Safari:

https://github.com/user-attachments/assets/b0cb0211-9471-487c-93b8-3cc5f43d435a

Firefox:

https://github.com/user-attachments/assets/bd8a0dcf-fa8e-4a99-a6f0-0b026ce88bdf

Chrome:

https://github.com/user-attachments/assets/28961369-29e1-4a94-bd33-6e74b97aeb6d

**Comment by greatislander** [comment]
The IDRC implemented a JavaScript-based reader using the [Web Speech Synthesis API](https://wicg.github.io/speech-api/#tts-section) on a recent project: https://github.com/accessibility-exchange/platform/pull/1376

**Comment by greatislander** [comment]
Also just to note that this does not resolve any WCAG violations but would be a usability enhancement.

**Comment by MichSpieler** [comment]
From meeting with Os: let's look into the IDRC implementation mentioned by Ned in a comment above. We think the current solutions offered by browsers should be good enough. What might be worth looking into is https://github.com/pressbooks/ideas/issues/530 to give authors the option to add pronunciation instructions.
